recipe-generator/src/main/resources/templates/js/hook.liquid

```diff
@@ -0,0 +1,21 @@
+{%- assign className = 'Abstract' | append: ingredient.name | append: 'Hook' -%}
+{%- assign ingredientSnapshotName = ingredient.name | append: 'Snapshot' -%}
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+
+const BaseIngredientHook = require("{{options.runtimeLibrary}}").BaseIngredientHook;
+const {{ingredientSnapshotName}} = require("./{{ingredient.name}}Data").{{ingredientSnapshotName}};
+
+class {{className}} extends BaseIngredientHook {
+    constructor() {
+        super("{{ingredient.name}}", {{ingredientSnapshotName}});
+        if (new.target === {{className}}) {
+            throw new TypeError("cannot instantiate {{className}} directly");
+        }
+    }
+}
+{%- for key in info.constantKeys %}
+{{className}}.{{key}} = "{{info.constantValues[forloop.index0]}}";
+{%- endfor %}
+
+exports.{{className}} = {{className}};
```

Comment:
If we're writing es6 classes and using const etc, why not use es6 import/export as well?

Reply:
I recall a while back when I was generating ts/js ingredients for protractor, node was having trouble loading ES6 imports, so I kept ES5 style loading. Maybe because we were using an older version of node to run protractor? But you're right, I should either go fully ES6 or ES5 here.

Reply:
I looked into this a bit more and it looks like nodejs doesn't have full support for ES6 modules yet and relies on the CommonJS style, but it does support other ES6 features, which explains why I can use `const` but still need `require`.

Reply:
I see 👍